### PR TITLE
Add fodjuomni to iset.mm

### PR DESCRIPTION
**Pull request by jkingdon**
This is Lemma 3.2 from [PradicBrown2021] and in their words/notation it is:

```
Suppose that we have an omniscient set O and some sets A and B. If there exists
a surjection f : O → A + B, then either A is inhabited or it is empty.
```

It is convenient to break this into a handful of lemmas but none of them are especially difficult.

### CODE REVIEW

File: iset.mm [hunk 1]
```diff
@@ -65603,6 +65603,112 @@ elements or fails to hold (is equal to ` (/) ` ) for some element.
       OXGWPVOSWQWTXASWRBWSXBXC $.
   $}
 
+  ${
+    $d A z $.  $d B w z $.  $d F w z $.  $d O w z $.  $d X w z $.  $d ph w z $.
+    fodjuomnilemdc.fo $e |- ( ph -> F : O -onto-> ( A |_| B ) ) $.
+    $( Lemma for ~ fodjuomni .  Decidability of a condition we use in various
+       lemmas.  (Contributed by Jim Kingdon, 27-Jul-2022.) $)
+    fodjuomnilemdc $p |- ( ( ph /\ X e. O )
+        -> DECID E. z e. A ( F ` X ) = ( inl ` z ) ) $=
```
Comment (benjub) [inline iset.mm:65612]: In this statement and the next one, I would use ``( ( inl |` A ) ` z )`` (since it's within a `E. z e. A`, it's equivalent). The reason is that  ``( inl |` A )`` is more tame than the raw `inl`, especially when `A` is a set. Up to you to see if it's worth it.

File: iset.mm [hunk 2]
```diff
@@ -65603,6 +65603,112 @@ elements or fails to hold (is equal to ` (/) ` ) for some element.
...
+          HVJZVQKTLWOVQOPZGQZRSZOIUAAVQVDVPWSMOIVNWQSVOWRRVNWQGVEUJVFVGVHVIAVSW
+          BAVSVBCDMEFGHIAWMVSJTAWPVSKTLAVSVDVKVIVLVM $.
+      $}
+    $}
+
+    ${
+      $d A a b x y z $.  $d B z $.  $d F a b y z $.  $d O a y z $.
+      $d ph y z $.
+      $( A condition which ensures ` A ` is either inhabited or empty.  Lemma
+         3.2 of [PradicBrown2021], p. 4.  (Contributed by Jim Kingdon,
```
Comment (benjub) [inline iset.mm:65701]: Can you phrase that condition ? E.g.: If..., then A is either inhabited of empty. (**edit**: it's basically what you wrote in your first comment above on the github PR page; I think it has its place as a comment in iset.mm)

File: iset.mm (hunk 1 above)
Comment (jkingdon) [reply iset.mm:65612]: I'm not sure whether to (again) try to describe my thoughts on this, or whether to just take "up to you" literally and say "I choose not to put in those restrictions".

File: iset.mm (hunk 2 above)
Comment (jkingdon) [reply iset.mm:65701]: I didn't write that, it is a quote from the paper.

My philosophy is that the formal statement of the theorem should (most of the time) be how we state what is being proved. Especially when proposed wording doesn't end up being any more concise or otherwise different from the formal notation.

File: iset.mm (hunk 1 above)
Comment (benjub) [reply iset.mm:65612]: I thought you were still ambivalent, but if that's settled, then ok. (As you noticed, I'm often bold in my comments in order to avoid ambiguity, and apart from the occasional obvious jokes, you are encouraged to take my comments literally.)

File: iset.mm (hunk 2 above)
Comment (benjub) [reply iset.mm:65701]: If you wrote the statement in plain English above (the quotation from the paper), it is because it helps.  So we both think it is helpful. Putting it in the comment in iset.mm would therefore be helpful.

**Review by benjub — APPROVED** [review]
